src/main/java/monitoringsystemturbo/controller/ConfirmExportController.java

```diff
@@ -0,0 +1,48 @@
+package monitoringsystemturbo.controller;
+
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import monitoringsystemturbo.presenter.ConfirmExportPresenter;
+
+import java.io.IOException;
+import java.util.List;
+
+public class ConfirmExportController {
+    private Stage primaryStage;
+    private ConfirmExportPresenter confirmExportPresenter;
+
+    public ConfirmExportController(Stage primaryStage) {
+        this.primaryStage = primaryStage;
+    }
+
+    public List<String> showConfirmationAndGetAppList(List<String> applications)
+            throws IOException {
+        FXMLLoader loader = new FXMLLoader();
+
+        loader.setLocation(this.getClass().getResource("/confirmExportView.fxml"));
+        Parent rootLayout = loader.load();
+
+        Stage dialogStage = new Stage();
+        dialogStage.setTitle("Confirm export");
+        dialogStage.initModality(Modality.WINDOW_MODAL);
+        dialogStage.initOwner(primaryStage);
+
+        confirmExportPresenter = loader.getController();
+        confirmExportPresenter.setPrimaryStage(dialogStage);
+        confirmExportPresenter.setApplicationsNames(applications);
+        confirmExportPresenter.start();
+
+        Scene scene = new Scene(rootLayout);
+        dialogStage.setScene(scene);
+        dialogStage.showAndWait();
+
+        return confirmExportPresenter.getApplicationsToExport();
+    }
+
+    public boolean getCancelValue() {
+        return confirmExportPresenter.wasCancelled();
```

Comment:
jeżeli są potrzebne 2 wartości, które są zapisane w confirmExportPresenter to lepiej już zwrócić ten właśnie obiekt, niż tworzyć 2 metodę, która nie wiadomo kiedy będzie potrzebna oraz może generować NullPointerException

Reply:
nie do końca się z tym zgadzam, gdyż zwrócenie całego obiektu wiąże się z tym, że udostępniamy na zewnątrz jego wszystkie funkcjonalności, które być może nie powinny być uruchamiane, a potencjalnie dalejmy taką możlwość. Moim zdaniem, lepiej zwracać więcej, prostszych informacji niż cały duży obiekt. NPE raczej nie ma tu znaczenia, bo równie dobrze może pojawić się w wyższej warstwie, ale flow naszej aplikacji i tak na to nie pozwala, więc to nie powinien być problem.

Reply:
Ogólnie ta komunikacja ze zwracaniem wartości nie do końca do mnie przemawia, wolałbym coś w rodzaju callbacków, ale ze względu na okoliczności i czas mogę przy tym pozostać :P